nix/project.nix

```diff
@@ -1,18 +1,21 @@
 { lib, config, sets, defaultSources, sources, ... }:
 let
   pkgs = config.nixpkgs.pkgs;
-  haskell-language-server-source = pkgs.fetchgit {
-    inherit (builtins.fromJSON (builtins.readFile ./haskell-language-server.json))
-      url sha256 fetchSubmodules
-      ;
-    deepClone = true; # Not too expensive for this repo; avoids some nasty errors.
-  };
+  haskell-language-server-source = pkgs.fetchgit
+    (builtins.removeAttrs (builtins.fromJSON (builtins.readFile ./haskell-language-server.json)) ["date"]);
   defaults = {
     configuration.packages.ghc.flags.ghci = lib.mkForce true;
     configuration.packages.ghci.flags.ghci = lib.mkForce true;
     configuration.reinstallableLibGhc = true;
     # This fixes a performance issue, probably https://gitlab.haskell.org/ghc/ghc/issues/15524
     configuration.packages.ghcide.configureFlags = [ "--enable-executable-dynamic" ];
+
+    # fixme: how to override a haskell.nix option?
```

Comment:
This is the correct way to define haskell.nix options, but previously I have noticed that they aren't used by the `stackProject` function.

Reply:
OK great. This configuration does seem to prevent at least one error when fetching the shake branch from git.